python-avd/pyavd/_eos_designs/structured_config/network_services/ethernet_interfaces.py

```diff
@@ -45,6 +45,22 @@ def ethernet_interfaces(self: AvdStructuredConfigNetworkServicesProtocol) -> Non
                     # to only contain entries with our hostname
                     self._set_l3_interfaces(vrf, tenant, subif_parent_interface_names)
 
+                    # Member ethernet ports for Port-Channel interface
+                    for l3_port_channel in vrf.l3_port_channels:
+                        # sub-interface for l3_port_channel cannot have member eth ports
+                        # skip any logic to generate member port config for such sub-interfaces
+                        if "." in l3_port_channel.name:
+                            continue
+                        member_eth_intfs = self._get_l3_port_channel_member_ports_cfg(l3_port_channel)
+                        for member_eth_intf in member_eth_intfs:
+                            append_if_not_duplicate(
+                                list_of_dicts=ethernet_interfaces,
+                                primary_key="name",
+                                new_dict=member_eth_intf,
+                                context=f"Ethernet interface defined under 'member_interfaces' for {self.shared_utils.node_type_key_data.key} l3_port_channels",
+                                context_keys=["name", "peer", "peer_interface"],
+                            )
```

Comment:
This should instead be adding the proper class items directly into self.structured_config.ethernet_interfaces, where we have builtin duplication checks.
Notice we have changed the l3_interfaces to just call a _set_l3_interfaces that sets it directly instead of returning here. Please try to update accordingly.

I know it is a moving target, as we are in the middle of a refactor, so don't hesitate to reach out if you need a hand on something.

Reply:
done